src/ApiService/ApiService/onefuzzlib/Metrics.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.OneFuzz.Service.OneFuzzLib.Orm;
+
+namespace Microsoft.OneFuzz.Service {
+
+    public record CustomMetric
+    (
+        string Name,
+        int Value,
+        DateTime CreatedAt,
+        BaseMetric CustomDimensions
+    );
+
+
+    public interface IMetrics {
+        Async.Task SendMetric(int metricValue, BaseMetric customDimensions);
+
+        void LogMetric(BaseMetric metric);
+    }
+
+    public class Metrics : IMetrics {
+        private readonly ILogTracer _log;
+        private readonly IOnefuzzContext _context;
+        private readonly JsonSerializerOptions _options;
+
+        public Metrics(ILogTracer log, IOnefuzzContext context) {
+            _context = context;
+            _log = log;
+            _options = new JsonSerializerOptions(EntityConverter.GetJsonSerializerOptions()) {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+            };
+            _options.Converters.Add(new RemoveUserInfo());
+        }
+
+        public async Async.Task QueueCustomMetric(MetricMessage message) {
```

Comment:
Can you explain the decision of queueing the metrics instead of logging it directly

Reply:
Just added a PR description that answers this!

In order to centralize and standardize the CustomMetric format, we are making use of a new storage queue custom-metrics and a new function QueueCustomMetrics in the app function. Funneling all of the logging through this one source will allow us to keep better track of the creation of the actual metric and, again, standardize the metric's format - name, value, custom dimensions.

Reply:
Added: 
In addition, this function app may act as a starting point for eventually _storing_ metrics and logging as we consider moving away from app insights entirely.